crypto/src/tx.rs

```diff
@@ -0,0 +1,126 @@
+use rand::seq::SliceRandom;
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{
+    addresses::PaymentAddress,
+    asset,
+    keys::{OutgoingViewingKey, SpendKey},
+    memo::MemoPlaintext,
+    merkle::proof::MerklePath,
+    Note, Output, Spend, Value,
+};
+
+/// Used to construct a Penumbra transaction.
+pub struct TransactionBuilder {
+    // Notes we'll consume in this transaction.
+    spends: Vec<Spend>,
+    // Notes we'll create in this transaction.
+    outputs: Vec<Output>,
+    // Transaction fee.
+    fee: u64,
+    // Total value changed in this transaction.
+    balance: i64,
+    // Put chain_id and anchor in here too?
+}
+
+impl TransactionBuilder {
+    pub fn new() -> Self {
+        Self {
+            spends: Vec::<Spend>::new(),
+            outputs: Vec::<Output>::new(),
+            fee: 0,
+            // xx Per asset?
+            balance: 0,
+        }
+    }
```

Comment:
I think this would be better as the body of an `impl Default`; we might not want `TransactionBuilder::new()` at all, and instead have a `Transaction::builder() -> TransactionBuilder { Default::default() }`, so that people don't need to import the builder type separately.